jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -540,33 +537,44 @@ public void updateInactivityTimer ()
     }
     
     /**
-     * Set the inactivity timer
+     * Set the inactivity timer on the session.
+     * 
+     * @param ms the interval in millisec 
      * 
-     * @param ms value in millisec, -1 disables it
      */
     private void setInactivityTimer (long ms)
     {
-        if (_sessionInactivityTimer == null)
-            _sessionInactivityTimer = new SessionInactivityTimeout();
         _sessionInactivityTimer.setIdleTimeout(ms);
     }
+    
+   
 
-
-    /**
-     * 
-     */
     public void stopInactivityTimer ()
+    {
+        stopInactivityTimer(false);
+    }
+
+    public void stopInactivityTimer (boolean destroy)
```

Comment:
I don't like the boolean arg.  I think `destroy` should be explicitly called elsewhere.